integration/tests/cook/test_basic.py

```diff
@@ -3075,6 +3087,8 @@ def test_pod_submission_failed(self):
         self.assertEqual('failed', job['instances'][0]['status'], job)
         self.assertEqual('Invalid task', job['instances'][0]['reason_string'], job)
 
+    @unittest.skipUnless(util.docker_image_for_default_pool() and not util.docker_image(),
+                         'We need a docker image to run this test, but dont have a pool from which to learn it.')
```

Comment:
> We need a docker image to run this test

Why?

Reply:
We have logic in the integration tests that tries to fill in a docker image onto everything. However, that logic assumes we have a pool specified (so that it can lookup the right docker image). When there's no pool, it fails to attach an image, and this test fails

Reply:
Shouldn't this test run just fine in environments where we're not using Docker? It currently does.

Reply:
Restructured how we test due to our conversation. Moot.